base/libgit2/callbacks.jl

```diff
@@ -59,6 +59,8 @@ function credentials_callback(cred::Ptr{Ptr{Void}}, url_ptr::Cstring,
     # parse url for schema and host
     urlparts = match(urlmatcher, url)
     schema = urlparts.captures[1]
+    urlusername = urlparts.captures[4]
```

Comment:
usually, `username` from url is `git` for `git+ssh` protocol. You can use `username_ptr` parameter to get it.